File: unity/ArabSolitaireUnity/Assets/ArabSolitaire/Bridge/BridgeMessageType.cs
using System;

namespace ArabSolitaire.Bridge
{
    public enum BridgeMessageType
    {
        StateSnapshot,
        TransitionResult,
        ActionIntent,
        UnityReady,
        RequestHint,
        RequestRestart,
        RequestExit,
        HintResult,
        ClientError,
    }

    public static class BridgeMessageTypeExtensions
    {
        public static string ToWireName(this BridgeMessageType type) => type switch
        {
            BridgeMessageType.StateSnapshot => "stateSnapshot",
            BridgeMessageType.TransitionResult => "transitionResult",
            BridgeMessageType.ActionIntent => "actionIntent",
            BridgeMessageType.UnityReady => "unityReady",
            BridgeMessageType.RequestHint => "requestHint",
            BridgeMessageType.RequestRestart => "requestRestart",
            BridgeMessageType.RequestExit => "requestExit",
            BridgeMessageType.HintResult => "hintResult",
            BridgeMessageType.ClientError => "clientError",
            _ => throw new ArgumentOutOfRangeException(nameof(type), type, null),
        };

        public static bool TryParse(string wireName, out BridgeMessageType type)
        {
            foreach (BridgeMessageType value in Enum.GetValues(typeof(BridgeMessageType)))
            {
                if (value.ToWireName() == wireName)
                {
                    type = value;
                    return true;
                }
            }

            type = default;
            return false;
        }
    }
}

File: packages/unity_bridge_contracts/lib/src/bridge_message_type.dart
/// Directional bridge message types (wire names match [name]).
enum BridgeMessageType {
  // Flutter / Dart → Unity
  initialize,
  loadLevel,
  stateSnapshot,
  transitionResult,
  hintResult,
  pause,
  resume,
  showStoryBeat,
  shutdown,
  fatalError,

  // Unity → Flutter / Dart
  unityReady,
  actionIntent,
  requestHint,
  requestRestart,
  requestExit,
  storyBeatSkipped,
  presentationCompleted,
  clientError;

  String get wireName => name;

  static BridgeMessageType parse(String raw) {
    for (final value in BridgeMessageType.values) {
      if (value.name == raw) return value;
    }
    throw FormatException('Unknown bridge message type: $raw');
  }
}
